device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfo.scala

```diff
@@ -0,0 +1,58 @@
+/**
+  * Copyright: Copyright (C) 2015, Jaguar Land Rover
+  * License: MPL-2.0
+  */
+package org.genivi.sota.device_registry
+
+
+import io.circe.Json
+import org.genivi.sota.data.Namespace
+import org.genivi.sota.device_registry.common.{Errors, SlickJsonHelper}
+import slick.driver.MySQLDriver.api._
+import org.genivi.sota.db.SlickExtensions._
+
+import scala.concurrent.ExecutionContext
+
+object GroupInfo extends SlickJsonHelper {
+
+  type GroupInfoType = Json
+  case class GroupInfo(groupName: String, namespace: Namespace, groupInfo: GroupInfoType)
+
+  // scalastyle:off
+  class GroupInfoTable(tag: Tag) extends Table[GroupInfo] (tag, "DeviceGroup") {
+    def groupName = column[String]("group_name", O.PrimaryKey)
+    def namespace = column[Namespace]("namespace", O.PrimaryKey)
+    def groupInfo = column[Json]("group_info")
+
+    def * = (groupName, namespace, groupInfo).shaped <>
+      ((GroupInfo.apply _).tupled, GroupInfo.unapply)
+  }
+  // scalastyle:on
+
+  val groupInfos = TableQuery[GroupInfoTable]
+
+  protected def getGroup(groupName: String, namespace: Namespace) =
+    groupInfos.filter(r => r.groupName === groupName && r.namespace === namespace)
+
+  def findGroupInfoByName(groupName: String, namespace: Namespace)(implicit ec: ExecutionContext)
+      : DBIO[GroupInfoType] = {
+    import org.genivi.sota.db.Operators._
+    getGroup(groupName, namespace)
+      .result
+      .headOption
+      .failIfNone(Errors.MissingGroupInfo)
+      .map(_.groupInfo)
+  }
+
+  def createGroupInfo(groupName: String, namespace: Namespace, data: GroupInfoType)(implicit ec: ExecutionContext)
+      : DBIO[Int] =
+    (groupInfos += GroupInfo(groupName, namespace, data))
+      .handleIntegrityErrors(Errors.GroupInfoAlreadyExists)
+
+  def updateGroupInfo(groupName: String, namespace: Namespace, data: GroupInfoType)(implicit ec: ExecutionContext)
```

Comment:
we dont need the `GroupInfo` part on these methods because whey will be called with `GroupInfo.update`